### PR TITLE
Evaluate schema.rb instead of parsing it like a string

### PR DESCRIPTION
**Pull request by mharris717**
Only todo is honor FORBIDDEN_FIELD_NAMES, although no such columns appear in our schema.rb. I diffed the output from this against the old output and it was identical.

### CODE REVIEW

**Comment by joshk0** [comment]
looks like a bunch of merge conflicts?

> We should produce schema.rb from rails 4 or rails 5 and confirm this continues to work

Sounds like a unit test to me :muscle:

**Comment by joshk0** [comment]
who should be reviewing this?

**Comment by roshan** [comment]
@tuliren

**Comment by mharris717** [comment]
I'll clean up the stuff we called out today. A for real today, not a fake today that takes 2 weeks like last time :(

**Comment by joshk0** [comment]
how's it going?

**Comment by tuliren** [comment]
I think there are still unresolved TODOs, like adding support for `FORBIDDEN_FIELD_NAMES` (and maybe run travisci on different gemfile versions as mentioned in above comments). Will take a look after those are done.

**Comment by joshk0** [comment]
@tuliren 
Since you are more familiar with the scenarios that Jack needs to be tested on, and the testing ecosystem, could you perhaps pair with @mharris717 on the remaining efforts? I am eager for this change to go in.

**Comment by tuliren** [comment]
Currently we don't have many test cases for the ruby part. All specs live in here:
https://github.com/LiveRamp/jack/tree/master/jack-test/test/rb

If I am going to write the spec, I would probably add test cases to make sure all the old features in `SchemaRbParser` are still functioning in the new parser.

Not sure there is anything else we can pair on. The code still looks incomplete, it's not ready for testing yet. But we can sync more about this on slack.

**Comment by tuliren** [comment]
> or why there are two gemfiles tbh

I have confirmed that the `Gemfile` and `Gemfile.lock` in [`jack-core/src`](https://github.com/LiveRamp/jack/tree/master/jack-core/src) are not actually used and can be deleted. Here is the [branch](https://github.com/tuliren/jack/tree/delete-core-gemfile). Feel free to delete them in this PR, unless you want me to merge mine (which may create more merge conflict).

**Comment by mharris717** [comment]
@tuliren PTAL

**Comment by tuliren** [comment]
@mharris717, thanks for the update. Can you add some rspecs in [jack-test/test/rb](https://github.com/LiveRamp/jack/tree/master/jack-test/test/rb)? This PR does not change anything in `db_schemas`, right?

**Comment by mharris717** [comment]
@tuliren it doesn't change anything in db_schemas. 

Gonna be honest with you. Normally, I'd be the first one to be screaming to add tests for some code that didn't have it. Here, I just don't have the energy right now to add tests to a project that currently has literally one in order to service my change. I diff'd the java code generated off master against what's generated by this, and it's identical. Is that enough for you right now?

**Comment by joshk0** [comment]
That's a perfectly fine approach. You've also explained the easiest test
you could write that would be really effective. Just a regression test that
compares the generated Java vs the checked in static version. That way
you're aware whenever the Java output changes, and more importantly, it's
obvious when it doesn't change.

If you don't feel like giving it a shot I think I will.

Thanks for all this! Let me know when I can sweep ruby 1.8 😀

File: jack-core/src/rb/schema_rb_parser.rb [hunk 1]
```diff
@@ -12,49 +12,118 @@
...
+        res[f.to_s] = v if !v.nil?
       end
+      res
     end
-    return models, migration_number
+  end
+end
+
+class SchemaRbParser
+  def self.parse(schema_rb, ignored_tables)
```
Comment (tuliren) [inline jack-core/src/rb/schema_rb_parser.rb]: `ignored_tables` is no longer optional, which breaks the main function (line 131, post your change).

Comment (tuliren) [reply jack-core/src/rb/schema_rb_parser.rb]: This is fixed in https://github.com/LiveRamp/jack/commit/a2617c8e26d8e0f11ae10fe92cc9dde7dff8f1d3.

**Comment by joshk0** [comment]
Thanks for this @tuliren . How would we know when this version of jack is in use throughout liveramp and we can sweep the old version of Ruby?

**Comment by tuliren** [comment]
Currently the redshift schema project is broken due to this change. I am working on a fix. Other than that, since jenkins build already succeeded for jack, I suppose it has been propagated to all projects company wide.

**Comment by tuliren** [comment]
`redshift_schemas` has been fixed. I am pretty sure the new version of jack is used everywhere now. But I don't know where the old ruby versions are used.

**Pull request merged**